metagraph/src/cli/query.cpp

```diff
@@ -484,16 +827,41 @@ int query_graph(Config *config) {
     return 0;
 }
 
-inline std::string query_sequence(size_t id,
-                                  const std::string &name,
-                                  const std::string &seq,
-                                  const AnnotatedDBG &anno_graph,
-                                  const Config &config) {
-    return QueryExecutor::execute_query(fmt::format_int(id).str() + '\t' + name, seq,
-                                        config.count_labels, config.print_signature,
-                                        config.suppress_unlabeled, config.num_top_labels,
-                                        config.discovery_fraction, config.anno_labels_delimiter,
-                                        anno_graph);
+inline void query_sequence(size_t id, std::string name, std::string seq,
+                           const AnnotatedDBG &anno_graph,
+                           const Config &config,
+                           const align::DBGAlignerConfig *aligner_config,
+                           const std::function<void(const std::string&)> &callback) {
+    if (aligner_config) {
+        auto matches = build_aligner(anno_graph.get_graph(), *aligner_config)->align(seq);
+        if (matches.size()) {
+            auto &match = matches[0];
```

Comment:
Does it always return multiple top matches, or we can tell the aligner that we want just the top alignment? Or the complexity is the same either way?

Reply:
You can tell the aligner how many top matches to return, but here we only take the top one. There's a check in `config` which tells the user about this and exits if they change the default number of returned alignments.

Reply:
So we can add an assertion here?